bootloader/bootloader_test.go

```diff
@@ -112,14 +113,98 @@ func (s *bootenvTestSuite) TestInstallBootloaderConfig(c *C) {
 		},
 		{name: "androidboot", gadgetFile: "androidboot.conf", sysFile: "/boot/androidboot/androidboot.env"},
 		{name: "lk", gadgetFile: "lk.conf", sysFile: "/boot/lk/snapbootsel.bin"},
-		{name: "grub recovery", gadgetFile: "grub-recovery.conf", sysFile: "/EFI/ubuntu/grub.cfg", opts: &bootloader.Options{Recovery: true}},
 	} {
 		mockGadgetDir := c.MkDir()
+		rootDir := c.MkDir()
 		err := ioutil.WriteFile(filepath.Join(mockGadgetDir, t.gadgetFile), t.gadgetFileContent, 0644)
 		c.Assert(err, IsNil)
-		err = bootloader.InstallBootConfig(mockGadgetDir, s.rootdir, t.opts)
+		err = bootloader.InstallBootConfig(mockGadgetDir, rootDir, t.opts)
 		c.Assert(err, IsNil, Commentf("installing boot config for %s", t.name))
-		fn := filepath.Join(s.rootdir, t.sysFile)
+		fn := filepath.Join(rootDir, t.sysFile)
 		c.Assert(fn, testutil.FilePresent, Commentf("boot config missing for %s at %s", t.name, t.sysFile))
 	}
 }
+
+func (s *bootenvTestSuite) TestInstallBootloaderConfigFromAssets(c *C) {
+	for _, t := range []struct {
+		name                string
+		gadgetFile, sysFile string
+		gadgetFileContent   []byte
+		assetContent        []byte
+		assetName           string
+		opts                *bootloader.Options
```

Comment:
since all of these test cases all use Recovery: True could you just always do that in the test code and not have to duplicate the opts definition for each test case?